pkg/csi/service/vanilla/controller.go

```diff
@@ -1354,10 +1360,198 @@ func (c *controller) ValidateVolumeCapabilities(ctx context.Context, req *csi.Va
 
 func (c *controller) ListVolumes(ctx context.Context, req *csi.ListVolumesRequest) (
 	*csi.ListVolumesResponse, error) {
+	start := time.Now()
 	ctx = logger.NewContextWithLogger(ctx)
 	log := logger.GetLogger(ctx)
-	log.Infof("ListVolumes: called with args %+v", *req)
-	return nil, logger.LogNewErrorCode(log, codes.Unimplemented, "listVolumes")
+	volumeType := prometheus.PrometheusUnknownVolumeType
+	if !commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.ListVolumes) {
+		return nil, logger.LogNewErrorCode(log, codes.Unimplemented, "List Volumes")
+	}
+	cfg, err := common.GetConfig(ctx)
+	if err != nil {
+		return nil, logger.LogNewErrorf(log, "failed to read config. Error: %+v", err)
+	}
+
+	// Get the global query limit
+	maxEntries := cfg.Global.QueryLimit
+	if req.MaxEntries != 0 {
+		maxEntries = int(req.MaxEntries)
+	}
+
+	listVolumesInternal := func() (*csi.ListVolumesResponse, string, error) {
+		log.Infof("ListVolumes: called with args %+v", *req)
```

Comment:
This is not needed. This will dump a lot of logs in file which is not needed at all. Also change this in WCP if Amruthi has added it.

Reply:
I changed it due to a comment from Chethan, since all the other APIs log it using Infof, so we want to be consistent.
LMK if I need to change it to Debugf.

Reply:
Please change to debugf.